src/MLOps.NET/IMLLifeCycleManager.cs

```diff
@@ -53,5 +54,13 @@ public interface IMLLifeCycleManager
         /// <param name="filePath">Absolute or relative path to the model</param>
         /// <returns></returns>
         Task UploadModelAsync(Guid runId, string filePath);
+
+        /// <summary>
+        /// Gets the best run for an experiment based on a metric for e.g "Accuracy"
+        /// </summary>
+        /// <param name="experimentId"></param>
+        /// <param name="metricName"></param>
+        /// <returns></returns>
+        IRun GetBestRun(Guid experimentId, string metricName);
```

Comment:
Shouldn't this be a part of `IExperiment`? 
It is not a lifecycle method.

Reply:
@shanranm that's correct, another way is for it to live on the Experiment depending on if you prefer light entities or a more DTO style entity. I think in this case it belongs on the lifecycle context mostly because this will be a method one calls at the end of a training run to determine if this run is good enough to bother uploading an artifact/model. If it's not better than anything previously we may want to save storage space and skip the upload step. I recognize that it's a matter of taste though :)

Reply:
If we are going with lighter entities then this looks good! 👍